### PR TITLE
Fix resize handles missing/broken (BL-11684)

### PR DESCRIPTION
**Pull request by sujeffreyl**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)

### CODE REVIEW

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 0 of 1 files reviewed, 1 unresolved discussion

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2941 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj34ToA-IkuXhA-UIg:-NFj34ToA-IkuXhA-UIh:b-qxy275) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2941)):*
> ```XML
> 
>                 if (handle instanceof HTMLElement) {
>                     // Overwrite the default zIndex of 90 provided in the inline styles of modified_libraries\jquery-ui\jquery-ui-1.10.3.custom.min.js
> ```

I'm a little iffy about why this file provides the zIndex inline.  I feel like it's bad. The inline style makes it so that we can't style it with CSS, no? Unless we spam important.  For example, the line style even overrides the custom CSS rule targeting ui-resizable-handles in themes\bloom-jqueryui-theme\jquery-ui-1.8.16.custom.css.
I feel like the zIndex:90 iss probably our custom modification, but I can't tell from the history.  The anti-synergy with jquery-ui-1.8.16.custom.css further reinforces my feeling.

I'm hesitant to tweak it for 5.4. I would be for considering removing the zIndex:90 from the custom JQuery file in Bloom 5.5.

Thoughts? Re-do for 5.4, do 2 separate fixes for 5.4 vs. 5.4, just keep this? Something else?

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 0 of 1 files reviewed, 2 unresolved discussions

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2942 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj4MlT1CWsS2GlQQFm:-NFj4MlT1CWsS2GlQQFn:b-n6r017) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2942)):*
> ```XML
>                 if (handle instanceof HTMLElement) {
>                     // Overwrite the default zIndex of 90 provided in the inline styles of modified_libraries\jquery-ui\jquery-ui-1.10.3.custom.min.js
>                     handle.style.zIndex = "1002"; // @bubbleCanvasZIndex + 1
> ```

I can't think of any way to reference the .less variable from the .ts file.  Anybody know a way?

**Review by andrew-polk — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @sujeffreyl)

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2941 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj34ToA-IkuXhA-UIg:-NFj6hUsBcikz8FhHo3W:b-3zikfq) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2941)):*
<details><summary><i>Previously, sujeffreyl wrote…</i></summary><blockquote>

I'm a little iffy about why this file provides the zIndex inline.  I feel like it's bad. The inline style makes it so that we can't style it with CSS, no? Unless we spam important.  For example, the line style even overrides the custom CSS rule targeting ui-resizable-handles in themes\bloom-jqueryui-theme\jquery-ui-1.8.16.custom.css.
I feel like the zIndex:90 iss probably our custom modification, but I can't tell from the history.  The anti-synergy with jquery-ui-1.8.16.custom.css further reinforces my feeling.

I'm hesitant to tweak it for 5.4. I would be for considering removing the zIndex:90 from the custom JQuery file in Bloom 5.5.

Thoughts? Re-do for 5.4, do 2 separate fixes for 5.4 vs. 5.4, just keep this? Something else?
</blockquote></details>

I'll let JT weigh in with the official opinion since he understands what is going on more, but here are my thoughts (which may just expose my ignorance):

1. We need to ship 5.4.
2. I don't understand your initial comment, "I'm a little iffy about why this file provides the zIndex inline." From what I see, the only place we set zindex in this file is in your new code here. Or are you just saying you're not sure you like your fix?
3. If the problem is putting this inline, can you just add a class with the rule instead?
___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2942 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj4MlT1CWsS2GlQQFm:-NFj5OW4CH7hBWAEJwsg:bcrnhx2) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2942)):*
<details><summary><i>Previously, sujeffreyl wrote…</i></summary><blockquote>

I can't think of any way to reference the .less variable from the .ts file.  Anybody know a way?
</blockquote></details>

No, but if it is important for them to be in sync, we should probably have comments in both places.

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @andrew-polk)

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2941 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj34ToA-IkuXhA-UIg:-NFj9ag26kofOBAe00Ko:b-jmy2eu) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2941)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I'll let JT weigh in with the official opinion since he understands what is going on more, but here are my thoughts (which may just expose my ignorance):

1. We need to ship 5.4.
2. I don't understand your initial comment, "I'm a little iffy about why this file provides the zIndex inline." From what I see, the only place we set zindex in this file is in your new code here. Or are you just saying you're not sure you like your fix?
3. If the problem is putting this inline, can you just add a class with the rule instead?


</blockquote></details>

Sorry, "this file" was amibiguous. It refers to the file mentioned in the line I highlighted. (That is, modified_libraries\jquery-ui\jquery-ui-1.10.3.custom.min.js)
___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2942 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj4MlT1CWsS2GlQQFm:-NFj9kpm5R1Axq3DfHIO:b5zpgtx) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2942)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

No, but if it is important for them to be in sync, we should probably have comments in both places.
</blockquote></details>

I added @bubbleCanvasZIndex in the comments here so that it could be find, but I'll add something more explicit to bloomUi.less too.

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @andrew-polk)

___
*[`src/BloomBrowserUI/bloomUI.less` line 36 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFjBPTRAF9rSuEGzqy4:-NFjBPTRAF9rSuEGzqy5:b17xwbt) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/04057f344e678666428d1a89d516bf721088246c/src/BloomBrowserUI/bloomUI.less#L36)):*
> ```Less
> @formatButtonZIndex: @textOverPictureZIndex + 1;
> @dragHandleEventZIndex: @textOverPictureZIndex + 1;
> @resizeHandleEventZIndex: @textOverPictureZIndex + 1; // Not directly referenced via .less, but conceptually used in bubbleManager.ts
> ```

This is actually 1003. (I was using 1002 in my Iteration 1). I just decided to make it consistent with @dragHandleEventZIndex, even though I suspect dragHandleEventZIndex could probably get away with 1002 instead of 1003.

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: all files reviewed (commit messages unreviewed), 2 unresolved discussions (waiting on @sujeffreyl)

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2941 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj34ToA-IkuXhA-UIg:-NFjBE1lCEb5op4EzVpN:b-6swamd) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2941)):*
<details><summary><i>Previously, sujeffreyl wrote…</i></summary><blockquote>

Sorry, "this file" was amibiguous. It refers to the file mentioned in the line I highlighted. (That is, modified_libraries\jquery-ui\jquery-ui-1.10.3.custom.min.js)
</blockquote></details>

Got it. Sorry I missed that.
So you have to put it inline here to override the inline there.
Very clear now that I see it.

We'll wait to see what JT says.

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 1 of 2 files reviewed, 3 unresolved discussions (waiting on @andrew-polk and @JohnThomson)

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 1043 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFni8Z-1IOQAP2rUNkZ:-NFnuC_d9YG9NPy96Bon:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/04057f344e678666428d1a89d516bf721088246c/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L1043)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

Getting a good name for isEventForEditableOrMoveHandle is getting harder, but I think it's basic purpose is to figure out that we've clicked on the text or some other control inside the bubble region that should 'do its normal thing' rather than dragging the bubble. So I'm thinking this logic should move there, with some suitable rename. At least, look at any other callers to see if they would want the fix too. Maybe isEventForEditableOrChildControl? isEventThatShouldNotDragBubble?
</blockquote></details>

Done.

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: 1 of 2 files reviewed, 3 unresolved discussions (waiting on @andrew-polk and @JohnThomson)

___
*[`src/BloomBrowserUI/bookEdit/js/bubbleManager.ts` line 2941 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505#-NFj34ToA-IkuXhA-UIg:-NFo-upA1LJgnZQroSe4:b-yerzoz) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/abb75020541e02a660e459b2b90b8cac1a48ba07/src/BloomBrowserUI/bookEdit/js/bubbleManager.ts#L2941)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

So, we think that before we touched anything, jquery was setting the z-index of drag handles in css and then overriding it in the style attribute? But we need a different one to put them in front of the canvas? That does seem ugly, but if it's how the library we're using works, I don't think what we're doing to work with it is ugly enough to try patching minified code.

We could try reducing @imageContainerZIndex to something less than the z-index that jquery uses (making sure to capture the reasons where it is declared) so that what jquery does just works. But messing with that feels risky for 5.4.

We can probably find the original jquery-ui\jquery-ui-1.10.3.min.js, assuming that's what our custom version is based on, to determine whether this nastiness is our change.

So, in summary, I think what Jeffrey has done here is the safest fix for 5.4.
For 5.5,
- if we can determine that messing with style for something normally set by css was our own patch, we should look for a better way.
- if it's part of how jquery draggable works, see if we can avoid the need to further override it by changing @imageContainerZIndex.

Another thought is that it may be time to consider upgrading to the latest jquery, or seeing if we can drop jquery altogether.
</blockquote></details>

Ok, so I downloaded jquery-ui @ 1.10.3. It's far from an identifical match for the diff, but I can clearly see in the 1.10.3 release that the zIndex=90 is from jquery-ui, not from us.

**Review by andrew-polk — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: all files reviewed, 2 unresolved discussions (waiting on @JohnThomson)

**Review by JohnThomson — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/5505)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sujeffreyl)
